File: frontend/lib/api/req/UpdateTaskReq.ts
import { TaskPriorityEnum } from "@api/dtos/TaskDto";

type UpdateTaskReq = {
  id: string;
  title: string;
  description?: string | null | undefined;
  dueDate?: string | null | undefined;
  isCompleted?: boolean | undefined;
  isImportant?: boolean | undefined;
  priority?: TaskPriorityEnum;
};

export type { UpdateTaskReq };

File: backend/Server/Contracts/Requests/UpdateTaskReq.cs
using Server.Contracts.Entities;

namespace Server.Contracts.Requests;

public class UpdateTaskReq
{
    public Guid Id { get; set; }
    public string Title { get; set; } = default!;
    public string? Description { get; set; }
    public DateOnly? DueDate { get; set; }
    public bool? IsCompleted { get; set; }
    public bool? IsImportant { get; set; }
    public TaskPriorityEnum Priority { get; set; }
}